modules/pipeline/pipengine/actionexecutor/plugins/docker/docker.go

```diff
@@ -0,0 +1,425 @@
+			NanoCPUs:   cpu.MilliValue(),
+		},
+		Privileged: true,
+	}
+	volMounts := d.GenerateDockerVolumes(job)
+	for _, mt := range volMounts {
+		hostCfg.Mounts = append(hostCfg.Mounts, mount.Mount{Type: mt.Type, Source: mt.Source, Target: mt.Target})
+	}
+	if job.PreFetcher != nil {
+		hostCfg.Mounts = append(hostCfg.Mounts, mount.Mount{Type: mount.TypeVolume, Source: d.makePreFetcherVolumeName(job), Target: job.PreFetcher.ContainerPath})
+	}
+	for _, bind := range job.Binds {
+		hostCfg.Mounts = append(hostCfg.Mounts, mount.Mount{Type: mount.TypeBind, Source: bind.HostPath, Target: bind.ContainerPath, ReadOnly: bind.ReadOnly})
+	}
+	con, err := d.client.ContainerCreate(ctx, &cfg, &hostCfg, nil, nil, d.makeJobName(job.Namespace, job.Name))
+	if err != nil {
+		return container.ContainerCreateCreatedBody{}, err
+	}
+	return con, nil
+}
+
+func (d *DockerJob) generatePreFetcherContainer(ctx context.Context, job *apistructs.JobFromUser) (container.ContainerCreateCreatedBody, error) {
+	jobImage, err := d.client.ImagePull(ctx, job.PreFetcher.FileFromImage, dockertypes.ImagePullOptions{})
+	if err != nil {
+		return container.ContainerCreateCreatedBody{}, errors.Errorf("failed to pull image, err: %v", err)
+	}
+	defer jobImage.Close()
+	preFetcherVolume, err := d.client.VolumeCreate(context.Background(), volume.VolumeCreateBody{
+		Name:   d.makePreFetcherVolumeName(job),
+		Driver: "local",
+	})
+	if err != nil {
+		return container.ContainerCreateCreatedBody{}, err
+	}
+	cfg := container.Config{
+		Image: job.PreFetcher.FileFromImage,
+		Env:   d.generateContainerEnvs(job),
+	}
+	hostCfg := container.HostConfig{
+		AutoRemove: true,
+		Privileged: true,
+		Mounts:     []mount.Mount{{Type: mount.TypeVolume, Source: preFetcherVolume.Name, Target: job.PreFetcher.ContainerPath}},
+	}
+	preFetcherContainer, err := d.client.ContainerCreate(context.Background(), &cfg, &hostCfg, nil, nil, d.makePreFetcherJobName(job))
+	if err != nil {
+		return container.ContainerCreateCreatedBody{}, err
+	}
+	return preFetcherContainer, nil
+}
+
+func (d *DockerJob) generateContainerEnvs(job *apistructs.JobFromUser) []string {
+	env := make([]string, 0)
+	envMap := job.Env
+
+	for k, v := range envMap {
+		env = append(env, fmt.Sprintf("%s=%s", k, v))
+	}
+
+	// add docker label
+	env = append(env, "IS_DOCKER=true")
+
+	// add namespace label
+	env = append(env, fmt.Sprintf("DICE_NAMESPACE=%s", job.Namespace))
+	env = append(env,
+		fmt.Sprintf("DICE_CPU_ORIGIN=%f", job.CPU),
+		fmt.Sprintf("DICE_MEM_ORIGIN=%f", job.Memory),
+		fmt.Sprintf("DICE_CPU_REQUEST=%f", job.CPU),
+		fmt.Sprintf("DICE_CPU_LIMIT=%f", job.CPU),
+		fmt.Sprintf("DICE_MEM_LIMIT=%f", job.Memory),
+	)
+
+	// add container TerminusDefineTag env
+	if len(job.TaskContainers) > 0 {
+		env = append(env, fmt.Sprintf("%s=%s", apistructs.TerminusDefineTag, job.TaskContainers[0].ContainerID))
+	}
+
+	// active upload logs
+	env = append(env, "ACTIONAGENT_ENABLE_PUSH_LOG_TO_COLLECTOR=true")
```

Comment:
ugly

Reply:
should move this to `executor env`, added annotation

Reply:
![image](https://user-images.githubusercontent.com/30427474/162690514-248d3ba7-381a-4bb5-b069-6b833fe07e4c.png)

Reply:
nono, you should search globally, and found this const inside agent pkg.

Reply:
adjusted, and modify other which use `ACTIONAGENT_ENABLE_PUSH_LOG_TO_COLLECTOR`